src/aiida/cmdline/commands/cmd_process.py

```diff
@@ -202,11 +219,14 @@ def process_call_root(processes):
     '-m', '--max-depth', 'max_depth', type=int, default=None, help='Limit the number of levels to be printed.'
 )
 @decorators.with_dbenv()
-def process_report(processes, levelname, indent_size, max_depth):
+def process_report(processes, most_recent_node, levelname, indent_size, max_depth):
     """Show the log report for one or multiple processes."""
     from aiida.cmdline.utils.common import get_calcjob_report, get_process_function_report, get_workchain_report
     from aiida.orm import CalcFunctionNode, CalcJobNode, WorkChainNode, WorkFunctionNode
 
+    if most_recent_node:
```

Comment:
I am surprised that there was no check here that the user provided at least one process.
Right now, running `verdi process report` just silently does nothing. Same comment for other subcommands. 

Compare with `verdi process play`
```console
$ verdi process play
Report: no active processes selected.
```
The log message however comes from `engine/processes/control.py` 
(should this be an Error instead of Report?)

Reply:
I agree this choice could be debated. Guess the logic was that since it takes a list, if an empty list is provided it behaves as any other for-loop in Python over an empty list, it is essentially a no-op. Definitely open to changing this and making it a warning or even error, but that is probably another PR.

Reply:
Sure. To be clear, I am less worried about the API of the functions in `control.py` and more about the CLI experience.

To me this is a really surprising behaviour (especially because the command is taking some time even with empty argument, but then it doesn't print anything).

I would still propose a change in this PR. If the most recent node was requested, we should error out if the user provided the process(es) argument at the same time.

Reply:
Done